open-api/rest-catalog-open-api.yaml

```diff
@@ -100,6 +121,16 @@ paths:
 
         Common catalog configuration settings are documented at
         https://iceberg.apache.org/docs/latest/configuration/#catalog-properties
+
+
+        Capabilities are used by the server to signal supported functionality and can be:
+        - tables (default capability in case the `capabilities` property doesn't exist or is empty in the response)
```

Comment:
The phrase in brackets gives the impression that all older service implementations do not support  views, signing, etc. This phrase breaks older service implementations accessed by clients that rely on this.

Reply:
the problem is that we don't know whether older service implementations do actually support views or not. I think we just need to agree what the default capability should be and based on earlier comments it was `tables`.
I'll start a ML thread on this topic so that we can discuss this properly

Reply:
Sorry if I missed the discussion, but assuming only tables still feels wrong.

Reply:
See https://github.com/apache/iceberg/pull/9940#issuecomment-2177231902 and https://github.com/apache/iceberg/pull/9940#issuecomment-2177235732 w.r.t using `tables` as the default capability in case the server doesn't send back any capabilities.

@snazy what other capability would you consider a default capability when an older server doesn't send back `capabilities`? I was considering adding `views` as a default as well but not every older server supports them.

Reply:
I've illustrated those on the ML: https://lists.apache.org/thread/c96xt4l4dnzgfor4bh5jjcl88gly3tm5

Reply:
I tend to think we ought to treat missing `capabilities` and empty `capabilities` elements differently.

If the element is present, that means the server is aware and declares its capabilities. There's no need to assume any defaults.

If the element is missing, I'd argue that the client should assume support for _all_ capabilities at the version level that was current when the client was built. Any possible errors that result from accessing unimplemented endpoints will have to be logged and debugged by the user, but that's always a possibility.

Reply:
@dimas-b I have updated the wording to make the expected behavior clearer